unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx

```diff
@@ -294,54 +342,75 @@ export const CreditCardForm = ({
     enabled: !!lock?.address,
   })
 
-  const [
-    { isLoading, data: isConnected = 0 },
-    { isLoading: isLoadingKeyGranter, data: keyGranter },
-  ] = useQueries({
-    queries: [
-      {
-        queryKey: [
-          'isStripeConnected',
-          lockAddress,
-          network,
-          disconnectStipeMutation.isSuccess,
-          connectStripeMutation.isSuccess,
-        ],
-        queryFn: isStripeConnected,
-      },
-      {
-        queryKey: ['getKeyGranter', lockAddress, network],
-        queryFn: getKeyGranter,
-      },
-    ],
-  })
-
   const isPricingLow = (fiatPricing?.usd?.amount ?? 0) < 0.5
 
   const loading = isLoading || isLoadingKeyGranter || isLoadingPricing
 
+  const onDisconnectStripe = async (event: any) => {
+    event.preventDefault()
+    const disconnectStripePromise = disconnectStipeMutation.mutateAsync()
+    await ToastHelper.promise(disconnectStripePromise, {
+      error: 'Stripe disconnection failed.',
+      success: 'Stripe disconnected.',
+      loading: 'Disconnecting Stripe.',
+    })
+    await refetchStripeConnectionDetails()
+  }
+
+  const onConnectStripe = (stripeAccount?: string) => {
+    connectStripeMutation.mutate(
+      { stripeAccount },
+      {
+        onSuccess: (connect: any) => {
+          if (connect?.url) {
+            window.location.assign(connect.url)
+          } else {
+            ToastHelper.success('Stripe connection succeeded!')
+          }
+        },
+        onError: () => {
+          ToastHelper.error('Stripe connection failed')
+        },
```

Comment:
we should stop relying on inline onError and onSuccess. They are going to be removed in react-query-v4